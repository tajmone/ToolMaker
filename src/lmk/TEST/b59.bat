..\lmk -lib .. b59.lmk > b59.tst.tmp 
move /Y stderr.txt b59.tst.win32 
type b59.tst.tmp >> b59.tst.win32 
cl -DWIN32 -o b59 b59List.c b59.c 
b59.exe -list unknown\unknown b59.c >> b59.tst.tmp 
type stderr.txt >> b59.tst.win32 
type b59.tst.tmp >> b59.tst.win32 
del b59.tst.tmp 
