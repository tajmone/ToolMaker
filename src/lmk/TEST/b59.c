/* b59.c			Date: 1995-11-29/toolmake

   b59 -- main program

   1.0 - 1995-11-29/
*/

#include <stdio.h>

#define PRIVATE static
#define PUBLIC


#include "b59List.h"


char *VERSION= "1.0";
char *NAME   = "b59";
char *USAGE= "Usage: b59 [-h] [-l <name>] <in> [<out>]";



/* -- DATA -- */

PRIVATE char 
    *listFileName= NULL,
    *inFileName= NULL,
    *outFileName= NULL;


/* -- SUBROUTINES -- */

/*----------------------------------------------------------------------
	perr()

	A simple writer of error messages (onto stderr),
	If 'addendum' is NULL it will not be written.
*/
PUBLIC void perr(
    char sev,
    char *desc,
    char *add
)
{
    if (add==NULL)
	fprintf(stderr, "%s: %c! %s\n", NAME, sev, desc);
    else
	fprintf(stderr, "%s: %c! %s: %s\n", NAME, sev, desc, add);

    if (sev=='F' || sev=='S') exit(1);
}


/* -- Argument processing -- */

PRIVATE void processArgs(
    int argc,
    char *argv[]
)
{				/* Go thru arguments, try to act on options */
    int i; 
 
    for (i=1; i<argc; ++i) {
        if (argv[i][0]=='-' && argv[i][1]) {
            switch (argv[i][1]) {
	    case 'l':
		listFileName= argv[++i];
		break;
	    case 'h':
		printf("%s\n\n", USAGE);
		printf("Options:\n");
		printf("  -l <name>  Set list file name\n");
		exit(0);
	    default:
		perr('W', "unknown switch", argv[i]);
		break;
	    }
	}
    	else if (inFileName==NULL) inFileName= argv[i];
    	else if (outFileName==NULL) outFileName= argv[i];
    	else perr('W', "unknown argument", argv[i]);
    }
}


/* -- MAIN -- */

PUBLIC main(
    int argc,
    char *argv[]
)
{
    int stdIn;    

#ifdef WIN32
    freopen("stderr.txt", "w", stderr);
#endif

    processArgs(argc, argv);
    printf("%s - %s\n\n", NAME, VERSION);

    if (inFileName == NULL)
        perr('F', "No input file", NULL);

    stdIn= strcmp(inFileName,"-")==0;
    if (stdIn)
	b59LiInit(VERSION, "standard input",  b59_ENGLISH_Messages);
    else
	b59LiInit(VERSION, inFileName,  b59_ENGLISH_Messages);
    /* .. try to open input file ... */
    if (0) {			/* Failed! */
        b59Log(NULL, 199, sevFAT, inFileName);
	b59List("", 0, 78, liTINY, sevALL);
    } else {
    	/* ... Parsing should be performed here ... */
        b59Log(NULL, 100, sevERR, NULL);
        if (stdIn) {		/* This gives an idea of own processing of
				   error messages */
	    int i;
	    char err[1024];
	    TmSrcp srcp;

	    /* Print a list on the terminal if errors detected. */
	    for (i=1; b59Msg(i, &srcp, err); i++)
		printf("\"-\", line %d: %s (column %d)\n",
		       srcp.line, err, srcp.col);
	} else {
	    /* Print a TINY list on the terminal */
	    b59List("", 0, 78, liTINY, sevALL);
	    /* And a list file if asked to. */
	    if (!stdIn && listFileName != NULL) {
		b59List(listFileName, 60, 132, liFULL, sevALL);
            }
	}
    }
    b59LiTerminate();
}
