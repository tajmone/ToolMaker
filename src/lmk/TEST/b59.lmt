%%SET lmkInclude(On)
%%SET lmkMessage(Embedded)
%%SET lmkListings(Single)
%%SET tmkForce(0)
%%SET tmkTarget("ansi-c")
%%SET lmkTarget("ansi-c")
%%SET tmkOs("SunOS")
%%SET lmkOs("SunOS")
%%SET tmkLibrary("%%(TMHOME)/lib/c")
%%SET lmkLibrary("..")
%%SET tmkPrefix("b59")
%%SET lmkPrefix("b59")
%%SET lmkEscape("`")
%%SET tmkSrcpName("TmSrcp")
%%SET tmkSrcpRow("line")
%%SET tmkSrcpCol("col")
%%SET tmkSrcpFile("file")
%%SET tmkSrcpDecl(" int line ", " int col ", " int file ")
%%SET lmkMSectName(%%(lmkPrefix)_ENGLISH_Messages)
%%SET lmkMSectOffs(0)
%%SET lmkMSectLen(10)
%%SET lmkMsgId("0     ")
%%SET lmkMsg("ToolMaker Lister 1.0")
%%SET lmkMsgId(, "1     ")
%%SET lmkMsg(, "No warnings or errors detected.")
%%SET lmkMsgId(, "2     ")
%%SET lmkMsg(, "No warnings issued.")
%%SET lmkMsgId(, "3     ")
%%SET lmkMsg(, "No errors detected.")
%%SET lmkMsgId(, "4     ")
%%SET lmkMsg(, "%1 informational message(s) produced.")
%%SET lmkMsgId(, "5     ")
%%SET lmkMsg(, "%1 warning(s) issued.")
%%SET lmkMsgId(, "6     ")
%%SET lmkMsg(, "%1 error(s) detected.")
%%SET lmkMsgId(, "7     ")
%%SET lmkMsg(, "Maximum number of messages exceeded.")
%%SET lmkMsgId(, "100   ")
%%SET lmkMsg(, "Message with one insert string: \""%1\""")
%%SET lmkMsgId(, "199   ")
%%SET lmkMsg(, "File \'%1\' could not be opened.")
%%PROCESS("%%(tmkLibrary)/Common.imp")
%%PROCESS("%%(lmkLibrary)/List.imp")
