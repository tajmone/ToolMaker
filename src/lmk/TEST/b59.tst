set os = "$argv[1]"
if ("$os" == "PC") then
  dos2unix -ascii b59.tst.win32
else
  ../lmk -lib .. b59.lmk
  gcc -o b59 b59List.c b59.c
  b59 -list /unknown b59.c
endif
