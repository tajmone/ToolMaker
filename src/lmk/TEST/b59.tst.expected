########## b59 ##########
set os = "$argv[1]"
if ("$os" == "PC") then
  dos2unix -ascii b59.tst.win32
else
  ../lmk -lib .. b59.lmk
  gcc -o b59 b59List.c b59.c
  b59 -list /unknown b59.c
endif
########## /bin/csh -f b59.tst UNIX ##########



        100 E : Message with one insert string: "%1"


1 error(s) detected.
No warnings issued.





b59.c

    1.  /* b59.c			Date: 1995-11-29/toolmake
    2.  
    3.     b59 -- main program
    4.  
    5.     1.0 - 1995-11-29/
    6.  */
    7.  
    8.  #include <stdio.h>
    9.  
   10.  #define PRIVATE static
   11.  #define PUBLIC
   12.  
   13.  
   14.  #include "b59List.h"
   15.  
   16.  
   17.  char *VERSION= "1.0";
   18.  char *NAME   = "b59";
   19.  char *USAGE= "Usage: b59 [-h] [-l <name>] <in> [<out>]";
   20.  
   21.  
   22.  
   23.  /* -- DATA -- */
   24.  
   25.  PRIVATE char 
   26.      *listFileName= NULL,
   27.      *inFileName= NULL,
   28.      *outFileName= NULL;
   29.  
   30.  
   31.  /* -- SUBROUTINES -- */
   32.  
   33.  /*----------------------------------------------------------------------
   34.  	perr()
   35.  
   36.  	A simple writer of error messages (onto stderr),
   37.  	If 'addendum' is NULL it will not be written.
   38.  */
   39.  PUBLIC void perr(
   40.      char sev,
   41.      char *desc,
   42.      char *add
   43.  )
   44.  {
   45.      if (add==NULL)
   46.  	fprintf(stderr, "%s: %c! %s\n", NAME, sev, desc);
   47.      else
   48.  	fprintf(stderr, "%s: %c! %s: %s\n", NAME, sev, desc, add);
   49.  
   50.      if (sev=='F' || sev=='S') exit(1);
   51.  }
   52.  


   53.  
   54.  /* -- Argument processing -- */
   55.  
   56.  PRIVATE void processArgs(
   57.      int argc,
   58.      char *argv[]
   59.  )
   60.  {				/* Go thru arguments, try to act on options */
   61.      int i; 
   62.   
   63.      for (i=1; i<argc; ++i) {
   64.          if (argv[i][0]=='-' && argv[i][1]) {
   65.              switch (argv[i][1]) {
   66.  	    case 'l':
   67.  		listFileName= argv[++i];
   68.  		break;
   69.  	    case 'h':
   70.  		printf("%s\n\n", USAGE);
   71.  		printf("Options:\n");
   72.  		printf("  -l <name>  Set list file name\n");
   73.  		exit(0);
   74.  	    default:
   75.  		perr('W', "unknown switch", argv[i]);
   76.  		break;
   77.  	    }
   78.  	}
   79.      	else if (inFileName==NULL) inFileName= argv[i];
   80.      	else if (outFileName==NULL) outFileName= argv[i];
   81.      	else perr('W', "unknown argument", argv[i]);
   82.      }
   83.  }
   84.  
   85.  
   86.  /* -- MAIN -- */
   87.  
   88.  PUBLIC main(
   89.      int argc,
   90.      char *argv[]
   91.  )
   92.  {
   93.      int stdIn;    
   94.  
   95.  #ifdef WIN32
   96.      freopen("stderr.txt", "w", stderr);
   97.  #endif
   98.  
   99.      processArgs(argc, argv);
  100.      printf("%s - %s\n\n", NAME, VERSION);
  101.  
  102.      if (inFileName == NULL)
  103.          perr('F', "No input file", NULL);
  104.  
  105.      stdIn= strcmp(inFileName,"-")==0;
  106.      if (stdIn)
  107.  	b59LiInit(VERSION, "standard input",  b59_ENGLISH_Messages);
  108.      else


  109.  	b59LiInit(VERSION, inFileName,  b59_ENGLISH_Messages);
  110.      /* .. try to open input file ... */
  111.      if (0) {			/* Failed! */
  112.          b59Log(NULL, 199, sevFAT, inFileName);
  113.  	b59List("", 0, 78, liTINY, sevALL);
  114.      } else {
  115.      	/* ... Parsing should be performed here ... */
  116.          b59Log(NULL, 100, sevERR, NULL);
  117.          if (stdIn) {		/* This gives an idea of own processing of
  118.  				   error messages */
  119.  	    int i;
  120.  	    char err[1024];
  121.  	    TmSrcp srcp;
  122.  
  123.  	    /* Print a list on the terminal if errors detected. */
  124.  	    for (i=1; b59Msg(i, &srcp, err); i++)
  125.  		printf("\"-\", line %d: %s (column %d)\n",
  126.  		       srcp.line, err, srcp.col);
  127.  	} else {
  128.  	    /* Print a TINY list on the terminal */
  129.  	    b59List("", 0, 78, liTINY, sevALL);
  130.  	    /* And a list file if asked to. */
  131.  	    if (!stdIn && listFileName != NULL) {
  132.  		b59List(listFileName, 60, 132, liFULL, sevALL);
  133.              }
  134.  	}
  135.      }
  136.      b59LiTerminate();
  137.  }
  138.  



        100 E : Message with one insert string: "%1"


1 error(s) detected.
No warnings issued.

b59 - 1.0

